src/cargo/ops/cargo_compile.rs

```diff
@@ -661,12 +661,28 @@ pub fn create_bcx<'a, 'cfg>(
                 continue;
             }
 
+            let guidance = if ws.is_ephemeral() {
+                if ws.ignore_lock() {
+                    "Try re-running cargo install with `--locked`".to_string()
+                } else {
+                    String::new()
+                }
+            } else {
+                format!(
+                    "Either upgrade to rustc {} or newer, or use\n\
+                     cargo update --precise -p {}@ver\n\
+                     where `ver` is the latest version of `{}` supporting rustc {}",
```

Comment:
This isn't the correct syntax for `cargo update`.  `--precise` requires a value. This is also passing the Display impl of `Package` which is not the correct syntax, either.

I think what this is looking for is something like `cargo update -p {name}@{vers} --precise VERSION` where `name = unit.pkg.name()` and `vers=unit.pkg.version()`. That's not perfect (in theory there could be conflicting `name@vers`, but that should be very rare).  Ideally it would use [`minimize`](https://github.com/rust-lang/cargo/blob/ba5b1920c4f0797bf30b30eb152265586c7f661e/src/cargo/core/package_id_spec.rs#L266) but that probably isn't too important.

Also, this advice is wrong for a workspace member. I think checking `unit.is_local()` should also be added here to skip the suggestion for `cargo update`.

Reply:
@ehuss I think I've fixed the first half of this.

Regarding workspaces, is that just because in that case you'd need to pass `--workspace`?

Reply:
No, the concern is that `cargo update` fundamentally won't give them a new version of a workspace member.

Imagine this scenario:
1. The `git2-rs` package has `rust-version = "1.64"` in its Cargo.toml.
2. Some random person runs `git clone https://github.com/rust-lang/git2-rs.git` to start hacking on it.
3. They have 1.62 installed. They run `cargo test`, and will immediately hit this error that `git2` doesn't support their version.
4. If the error recommends that they run `cargo update -p git2`, that has no effect.  What they actually need to do is get a newer rustc, or `git checkout` an older version (and I don't think cargo should recommend running `git` commands).

Reply:
Ah, I see! I thought you meant "if the package whose dependency won't compile in Cargo.toml is in a workspace"; I didn't realize you meant "if the package that won't compile is *itself* in a workspace".

I can fix that.